cpp/src/cylon/compute/aggregates.cpp

```diff
@@ -253,6 +252,192 @@ cylon::Status Max(const std::shared_ptr<cylon::Table> &table, int32_t col_idx, s
   return MinMax<1>(table, col_idx, output);
 }
 
+template<typename arrow_type, typename = typename std::enable_if<arrow::is_number_type<arrow_type>::value
+                                                                     | arrow::is_boolean_type<arrow_type>::value>::type>
+arrow::Status ResolveTableFromScalar(arrow::MemoryPool *pool,
+                                     std::shared_ptr<cylon::CylonContext> &ctx,
+                                     int32_t col_idx,
+                                     std::shared_ptr<cylon::compute::Result> &result,
+                                     std::shared_ptr<cylon::Table> input,
+                                     std::shared_ptr<cylon::Table> &output) {
+  using SCALAR_TYPE = typename arrow::TypeTraits<arrow_type>::ScalarType;
+  using IDX_BUILDER_T = typename arrow::TypeTraits<arrow_type>::BuilderType;
+
+  arrow::Status s;
+  std::vector<std::shared_ptr<arrow::Array>> out_vectors;
+  IDX_BUILDER_T idx_builder(pool);
+  std::shared_ptr<arrow::Array> out_idx;
+  if (!(s = idx_builder.Reserve(1)).ok()) {
+    return s;
+  }
+  std::shared_ptr<SCALAR_TYPE>
+      &&cast_sclr = std::static_pointer_cast<SCALAR_TYPE>(result->GetResult().scalar());
+
+  idx_builder.UnsafeAppend(cast_sclr->value);
+
+  if (!(s = idx_builder.Finish(&out_idx)).ok()) {
+    return s;
+  }
+
+  out_vectors.push_back(out_idx);
+
+  std::shared_ptr<arrow::Table> arw_table;
+  input->ToArrowTable(arw_table);
+
+  std::vector<std::shared_ptr<arrow::Field>> fields;
+  std::shared_ptr<arrow::Field> selected_filed = arw_table->schema()->field(col_idx);
+
+  fields.push_back(std::make_shared<arrow::Field>(selected_filed->name(),
+                                                  selected_filed->type(),
+                                                  selected_filed->nullable()));
+
+  auto schema = arrow::schema(fields);
+  auto out_a_table = arrow::Table::Make(schema, {out_vectors});
```

Comment:
I think this could be simplified like this (since there's only one element in the every vector)
```c++
 auto schema = arrow::schema({arw_table->schema()->field(col_idx)});
 auto out_a_table = arrow::Table::Make(schema, {out_index});
```

I don't think a separate array vector and a field vector is needed in this situation.